project/project.py

```diff
@@ -0,0 +1,156 @@
+        that resembles the same output from the Open Meteo API.
+        It then puts the data into a 2D array with a marker to signify
+        that it is a mock data set.
+        '''
+        data = '''{
+        "__CLASS__": "Adzuna::API::Response::JobSearchResults",
+        "results": [
+            {
+            "salary_min": 50000,
+            "longitude": -0.776902,
+            "location": {
+                "__CLASS__": "Adzuna::API::Response::Location",
+                "area": [
+                "UK",
+                "South East England",
+                "Buckinghamshire",
+                "Marlow"
+                ],
+                "display_name": "Marlow, Buckinghamshire"
+            },
+            "salary_is_predicted": 0,
+            "description": "JavaScript Developer Corporate ...",
+            "__CLASS__": "Adzuna::API::Response::Job",
+            "created": "2013-11-08T18:07:39Z",
+            "latitude": 51.571999,
+            "redirect_url": "http://adzuna.co.uk/jobs/land/ad/129698749...",
+            "title": "Javascript Developer",
+            "category": {
+                "__CLASS__": "Adzuna::API::Response::Category",
+                "label": "IT Jobs",
+                "tag": "it-jobs"
+            },
+            "id": "129698749",
+            "salary_max": 55000,
+            "company": {
+                "__CLASS__": "Adzuna::API::Response::Company",
+                "display_name": "Corporate Project Solutions"
+            },
+            "contract_type": "permanent"
+            },
+            ... another 19 ads here ...
+        ]
+        }
+        '''
+        link = "http://adzuna.co.uk/jobs/land/ad/129698749..."
+        return link
+
+class APIService(Service):
+    def get_job_data(self):
+        '''
+        This method asks for user input of the location of where they would
+        like to get temperature data from and then makes an API request call for weather and time in that location.
+        It puts the data into their respective arrays.
+        Finally, puts the data into a 2D array with a marker to signify
+        that it is an API data set.
+        '''
+        what = input("Please enter the job's title: ")
+        where = input("Please enter a preffered location: ")
+        salary_min = input("Please enter your minimum salary: ")
+
+        url = "https://api.adzuna.com/v1/api/jobs/us/search/1"
+        params = {
+            "app_id": "a37cffb8",  # replace with your app ID
+            "app_key": "7b817d31493e87cf2039cdf3cf24e06e",  # replace with your API key
+            "results_per_page": 1,
+            "what": what,
+            "where": where,
+            "salary_min": salary_min
+        }
+
+        req = requests.get(url, params=params)
+        response = req.json()
+        #print(response.json())
+            #can display pnly the link to the jobs
+        
+        if req.status_code == 200:
+            link = response[1][7]
```

Comment:
Based on the query response, there are few things you are doing incorrect here. Essentially, I think there is confusion about how to access values in a dictionary v/s a list. For a dictionary, values are accessed using their string 'key' values, and in a list, they are accessed by their 'index' integer. When you see `KeyError` messages, you should look to see what is the misalignment.

In your case, if you take a look at the response, anything that's enclosed in a `{}` is a dictionary type, and `[]` is a list type.

So for your response, we need to first access the key `'results'` in your dictionary. That key has a list of 'results' and to access that you need a index. Let's say you just care about the first one, you need `[0]` and then within that there is another nested dictionary to get the 'redirect_url' (or anything else you care about)

So to access the redirect url, it would look something like `link = response['results'][0]['redirect_url']`

Of course, if you are expecting to read more than just the first result, you will need to look through all the results, and use the data accordingly.